dashboard/app/models/user.rb

```diff
@@ -844,16 +848,28 @@ def User.track_level_progress_sync(user_id, level_id, script_id, new_result, sub
       user_level.save!
     end
 
-    if new_level_completed && script_id
+    if pairing_user_ids
+      pairing_user_ids.each do |navigator_user_id|
+        navigator_user_level = User.track_level_progress_sync(navigator_user_id, level_id, script_id, new_result, submitted)
+        retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
+          PairedUserLevel.find_or_create_by(navigator_user_level_id: navigator_user_level.id,
+                                            driver_user_level_id: user_level.id)
```

Comment:
It won't, `find_or_create` is not atomic (it's the same as calling `find` and then `create`) so we need to retry it (or wrap it in a transaction -- but retrying is usually better for perf), similar to why we need to retry the UserLevel first_or_create above.